offchain/dispatcher/src/signer/aws_signer.rs

```diff
@@ -149,3 +124,17 @@ impl Signer for AwsSigner {
         }
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    use rusoto_core::Region;
+
+    use crate::signer::aws_signer::AwsSigner;
+
+    #[tokio::test]
+    async fn new_aws_signer_with_error() {
+        let aws_signer =
+            AwsSigner::new("invalid".to_string(), 0, Region::UsEast1).await;
+        assert!(aws_signer.is_err());
```

Comment:
The error happening here is because the key is invalid?

Reply:
Correct! Added a commentary.